server/plugin/api.go

```diff
@@ -965,6 +966,33 @@ func (p *Plugin) getPrByNumber(w http.ResponseWriter, r *http.Request, userID st
 	p.writeJSON(w, result)
 }
 
+func (p *Plugin) searchLabels(w http.ResponseWriter, r *http.Request, userID string) {
+	info, apiErr := p.getGitHubUserInfo(userID)
+	if apiErr != nil {
+		p.writeAPIError(w, apiErr)
+		return
+	}
+
+	repo := r.URL.Query().Get("repo")
+	query := r.URL.Query().Get("q")
+	repoID, err := strconv.ParseInt(repo, 10, 64)
+	if err != nil {
+		mlog.Error(err.Error())
+		p.writeAPIError(w, &APIErrorResponse{Message: "Invalid Repo ID", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	githubClient := p.githubConnect(*info.Token)
+	result, _, err := githubClient.Search.Labels(context.Background(), repoID, query, &github.SearchOptions{})
```

Comment:
I couldn't find a way to consume the following endpoint using the `go-github` library:
https://docs.github.com/en/rest/reference/issues#list-labels-for-a-repository

It would make things much simpler if we were able to fetch all available labels instead of searching by query string.